command/genesis/params.go

```diff
@@ -121,24 +132,30 @@ func (p *genesisParams) areValidatorsSetByPrefix() bool {
 }
 
 func (p *genesisParams) getRequiredFlags() []string {
-	return []string{
-		command.BootnodeFlag,
+	if p.isIBFTConsensus() {
+		return []string{
+			command.BootnodeFlag,
+		}
 	}
+
+	return []string{}
 }
 
 func (p *genesisParams) initRawParams() error {
 	p.consensus = server.ConsensusType(p.consensusRaw)
 
-	if err := p.initIBFTValidatorType(); err != nil {
-		return err
-	}
+	if p.consensus != server.PolyBFTConsensus {
+		if err := p.initIBFTValidatorType(); err != nil {
+			return err
+		}
 
-	if err := p.initValidatorSet(); err != nil {
-		return err
-	}
+		if err := p.initValidatorSet(); err != nil {
+			return err
+		}
 
-	p.initIBFTExtraData()
-	p.initConsensusEngineConfig()
+		p.initIBFTExtraData()
+		p.initConsensusEngineConfig()
+	}
 
 	return nil
 }
```

Comment:
You can have a condition in the beginning and return if the consensus is not PolyBFT:

ex
```go
if p.isPolyBFTConsensus() {
   return nil
}
```